soc/arm/st_stm32/stm32l4/power.c

```diff
@@ -26,15 +26,34 @@ LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
 #define RCC_STOP_WAKEUPCLOCK_SELECTED LL_RCC_STOP_WAKEUPCLOCK_HSI
 #endif
 
-/* Invoke Low Power/System Off specific Tasks */
-void pm_power_state_set(struct pm_state_info info)
+static void pm_state_set_soft_off(uint8_t substate_id)
 {
-	if (info.state != PM_STATE_SUSPEND_TO_IDLE) {
-		LOG_DBG("Unsupported power state %u", info.state);
-		return;
+	switch (substate_id) {
+	case 1: /* this corresponds to the Shutdown mode: */
+		/* Disable Pin 1 which is used as wakeup source */
+		LL_PWR_DisableWakeUpPin(LL_PWR_WAKEUP_PIN1);
```

Comment:
This needs to be configurable. Can you check how to configure this from device tree ?

Reply:
Can you please let me know what exactly you wanted to be configured or any reference point?
I tried searching for the resources related to this pin configuration but haven't  found anything.

Reply:
ok, so I haven't found anything quick and easy to implement.
Mostly it would require preliminary changes in device tree that would take some time to think, implement and finally get merged.
This being said, I'd like to avoid merging code that only fit a particular implementation.

I'm thinking to an alternate proposal that would each one to test this and adapt easily on his use case.
Would you mind:
-Removing the code part that handle pin in this file
-Add a dedicated power_off sample under `samples/boards/stm32/power_mgmt` that would demonstrate use of one convenient PIN as a wake up. Ideally the sample would work on a in tree board.